### PR TITLE
[spirv] Call SPIR-V legalization passes from SPIRV-Tools

### PR DESCRIPTION
**Pull request by antiagainst**
When seeing opaque types within structs in function parameter,
function return, and variable definition, invoke SPIRV-Tools
legalization passes.

Also refreshed external projects

### CODE REVIEW

**Comment by antiagainst** [comment]
@ehsannas: PTAL

**Comment by antiagainst** [comment]
I've signed the CLA again. Hope this time the bot is happy and remembers me. ;-P

File: tools/clang/lib/SPIRV/TypeTranslator.cpp [hunk 1]
```diff
@@ -61,6 +61,53 @@ bool TypeTranslator::isRelaxedPrecisionType(QualType type) {
   return false;
 }
 
+bool TypeTranslator::isOpaqueType(QualType type) {
+  if (const auto *recordType = type->getAs<RecordType>()) {
+    const auto name = recordType->getDecl()->getName();
+
+    if (name == "Texture1D" || name == "RWTexture1D")
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/TypeTranslator.cpp]: Why are you not combining all these cases into one conditional? (or use a switch?)

Comment (antiagainst) [reply tools/clang/lib/SPIRV/TypeTranslator.cpp]: Just copied what the [existing code](https://github.com/Microsoft/DirectXShaderCompiler/blob/11dd70c4105012313f6e2d0c1a949ad4e94078d4/tools/clang/lib/AST/HlslTypes.cpp#L342) base does. I think it's more easier to read this way. Actually a `StringSwitch` seems nice for this. (We cannot use C `switch` on strings.)

**Pull request merged**